runners/core-java/src/test/java/org/apache/beam/runners/core/metrics/MetricsPusherTest.java

```diff
@@ -48,18 +48,35 @@
   private static final Logger LOG = LoggerFactory.getLogger(MetricsPusherTest.class);
 
   private static final long NUM_ELEMENTS = 1000L;
+  private static final String COUNTER_NAME = "counter";
   @Rule public final TestPipeline pipeline = TestPipeline.create();
 
   @Before
   public void init() {
-    TestMetricsSink.clear();
     MetricsOptions options = pipeline.getOptions().as(MetricsOptions.class);
     options.setMetricsSink(TestMetricsSink.class);
   }
 
   @Category({ValidatesRunner.class, UsesAttemptedMetrics.class, UsesCounterMetrics.class})
   @Test
-  public void test() throws Exception {
+  public void pushesUserMetrics() throws Exception {
+    TestMetricsSink.clear();
+    pipeline
+        .apply(
+            // Use maxReadTime to force unbounded mode.
+            GenerateSequence.from(0).to(NUM_ELEMENTS).withMaxReadTime(Duration.standardDays(1)))
+        .apply(ParDo.of(new CountingDoFn()));
+    pipeline.run();
+    // give metrics pusher time to push
+    Thread.sleep(
+        (pipeline.getOptions().as(MetricsOptions.class).getMetricsPushPeriod() + 1L) * 1000);
```

Comment:
We should probably lower this interval and build in a retry logic. Otherwise this is prone to breaking.